src/mqt/predictor/ml/helper.py

```diff
@@ -28,42 +24,15 @@ def qcompile(
     """Compiles a given quantum circuit to a device with the highest predicted figure of merit.
 
     Arguments:
-        qc: The quantum circuit to be compiled.
+        qc: The quantum circuit to be compiled
         figure_of_merit: The figure of merit to be used for compilation. Defaults to "expected_fidelity".
 
     Returns:
         Returns a tuple containing the compiled quantum circuit, the compilation information and the name of the device used for compilation. If compilation fails, False is returned.
     """
-    device = predict_device_for_figure_of_merit(qc, figure_of_merit)
-    res = rl.qcompile(qc, figure_of_merit=figure_of_merit, device_name=device.name)
-    return *res, device.name
-
-
-def predict_device_for_figure_of_merit(
-    qc: Path | QuantumCircuit, figure_of_merit: reward.figure_of_merit = "expected_fidelity"
-) -> Device:
-    """Returns the name of the device with the highest predicted figure of merit that is suitable for the given quantum circuit.
-
-    Arguments:
-        qc: The quantum circuit to be compiled.
-        figure_of_merit: The figure of merit to be used for compilation. Defaults to "expected_fidelity".
-
-    Returns:
-        The device with the highest predicted figure of merit that is suitable for the given quantum circuit.
-    """
-    ml_predictor = ml.Predictor()
-    predicted_device_index_probs = ml_predictor.predict_probs(qc, figure_of_merit)
-    assert ml_predictor.clf is not None
-    classes = ml_predictor.clf.classes_  # type: ignore[unreachable]
-    predicted_device_index = classes[np.argsort(predicted_device_index_probs)[::-1]]
-
-    num_qubits = qc.num_qubits if isinstance(qc, QuantumCircuit) else QuantumCircuit.from_qasm_file(qc).num_qubits
-
-    for index in predicted_device_index:
-        if ml_predictor.devices[index].num_qubits >= num_qubits:
-            return ml_predictor.devices[index]
-    msg = "No suitable device found."
-    raise ValueError(msg)
+    predicted_device = ml.predict_device_for_figure_of_merit(qc, figure_of_merit=figure_of_merit)
+    res = rl.qcompile(qc, figure_of_merit=figure_of_merit, device_name=predicted_device.name)
+    return *res, predicted_device
```

Comment:
according to codecov, these lines are not covered by any tests.
given how central this method is for the predictor, I would argue that this should not be the case and a corresponding test should be added.

Reply:
I added a new test for this as well. Really nice to see, that we now cover the whole Predictor flow :)

Reply:
Indeed 😎